docs/source/common/weights_loading.rst

```diff
@@ -41,83 +43,70 @@ To change the checkpoint path pass in:
+        trainer = Trainer(callbacks=[checkpoint_callback])
+        trainer.fit(model)
+        checkpoint_callback.best_model_path
 
 
-    # 3. Init ModelCheckpoint callback, monitoring 'val_loss'
-    checkpoint_callback = ModelCheckpoint(monitor="val_loss")
+Manual Saving
+=============
+You can manually save checkpoints and restore your model from the checkpointed state using :meth:`~pytorch_lightning.trainer.trainer.Trainer.save_checkpoint`
+and :meth:`~pytorch_lightning.core.saving.ModelIO.load_from_checkpoint`.
 
-    # 4. Add your callback to the callbacks list
-    trainer = Trainer(callbacks=[checkpoint_callback])
+.. code-block:: python
 
-You can also control more advanced options, like `save_top_k`, to save the best k models and the `mode` of the monitored quantity (min/max), `save_weights_only` or `every_n_epochs` to set the interval of epochs between checkpoints, to avoid slowdowns.
+    model = MyLightningModule(hparams)
+    trainer.fit(model)
+    trainer.save_checkpoint("example.ckpt")
+    new_model = MyLightningModule.load_from_checkpoint(checkpoint_path="example.ckpt")
 
-.. testcode::
 
-    from pytorch_lightning.callbacks import ModelCheckpoint
+Manual Saving with Distributed Training Strategies
+==================================================
 
+Lightning also handles strategies where multiple processes are running, such as DDP. For example, when using the DDP strategy our training script is running across multiple devices at the same time.
+Lightning automatically ensures that the model is saved only on the main process, whilst other processes do not interfere with saving checkpoints. This requires no code changes as seen below:
 
-    class LitAutoEncoder(LightningModule):
-        def validation_step(self, batch, batch_idx):
-            x, y = batch
-            y_hat = self.backbone(x)
-            loss = F.cross_entropy(y_hat, y)
-            self.log("val_loss", loss)
+.. code-block:: python
 
+    trainer = Trainer(strategy="ddp")
+    model = MyLightningModule(hparams)
+    trainer.fit(model)
+    # Saves only on the main process
+    trainer.save_checkpoint("example.ckpt")
 
-    # saves a file like: my/path/sample-mnist-epoch=02-val_loss=0.32.ckpt
-    checkpoint_callback = ModelCheckpoint(
-        monitor="val_loss",
-        dirpath="my/path/",
-        filename="sample-mnist-{epoch:02d}-{val_loss:.2f}",
-        save_top_k=3,
-        mode="min",
-    )
+Not using :meth:`~pytorch_lightning.trainer.trainer.Trainer.save_checkpoint` can lead to unexpected behavior and potential deadlock. Using other saving functions will result in all devices attempting to save the checkpoint. As a result, we highly recommend using the Trainer's save functionality.
+If using custom saving functions cannot be avoided, we recommend using the :func:`~pytorch_lightning.utilities.distributed.rank_zero_only` decorator to ensure saving occurs only on the main process.
```

Comment:
For model parallel use cases, using `rank_zero_only` will not work. `rank_zero_only` is recommended only if all ranks hold the exact same state.

Reply:
updated!